### PR TITLE
Add initial draft of python library

### PR DESCRIPTION
**Pull request by Others**
This is the first draft of the python library / associated test component

### CODE REVIEW

File: python/test_component.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+def test(http_method, path, request_arguments, request_body):
+    print(http_method, path, request_arguments, request_body)
+    return 200, "all good"
+
+
+if __name__ == '__main__':
+    print("Arguments " + str(sys.argv))
+    comp = V9Component(sys.argv[1], sys.argv[2])
+
+    comp.register_operation("Test", test)
```
Comment (hjaensch7) [inline python/test_component.py:14]: Can a component have multiple operations?

File: python/v9.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            except Exception as e:
+                return serialize_response(500, "User function failed " + repr(e))
+
+            return serialize_response(response_code, response_body)
+        except KeyError:
+            # 543 = internal error
+            # (This means the node has messed up serializing the line)
+            return serialize_response(543, "", internal_error="Invalid request " + deserialized_line)
+
+    def loop(self):
```
Comment (hjaensch7) [inline python/v9.py:59]: Is there anyway to make this less nested, because it's pretty confusing

File: python/test_component.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import sys
+from v9 import V9Component
+
+
+def test(http_method, path, request_arguments, request_body):
```
Comment (hjaensch7) [inline python/test_component.py:5]: Does this design allow for a developer to edit/populate the response to the request?

Comment (Others) [reply python/test_component.py:5]: Operations return a `response_code, response_body` tuple. Is that what you mean?

I will add documentation

File: python/test_component.py (hunk 1 above)
Comment (Others) [reply python/test_component.py:14]: Yes, you just need to call `register_operation` multiple times. For simplicity the example component just has one -- should I add another?

File: python/test_component.py (hunk 3 above)
Comment (hjaensch7) [reply python/test_component.py:5]: Oh yes, that makes sense. Adding documentation would be good.

File: python/v9.py (hunk 2 above)
Comment (Others) [reply python/v9.py:59]: There really isn't much actual code here. I probably could remove the inner try/except. That's still 4 levels of depth though... What part here do you find confusing specifically?

File: python/test_component.py (hunk 1 above)
Comment (hjaensch7) [reply python/test_component.py:14]: No need to add another, just wanted to confirm that a component can support multiple functions.

**Review by hjaensch7 — APPROVED** [review]

File: python/test_component.py (hunk 1 above)
Comment (Others) [reply python/test_component.py:14]: Also as a note, I plan to add opt-in automatic JSON desearlizing/serializing in the next draft

File: python/v9.py (hunk 2 above)
Comment (hjaensch7) [reply python/v9.py:59]: Mostly just complaining about the nested indents. What happens if the loop starts but the pipe is already empty?

Comment (Others) [reply python/v9.py:59]: It doesn't matter if there is stuff in the pipe or not -- either we block waiting on read, we fail a read, or we read a line. None of those three cases is handled erroneously as far as I can see?

Comment (hjaensch7) [reply python/v9.py:59]: Got it, the block happens in read, I didn't realize that.

**Pull request merged**